### PR TITLE
Added Shell Sort in C++

### LINKED ISSUE #4066: Shell Sort in C++
### Description

In this, we will get a brief idea about Shell Sort in C++

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by aniket-24**
<hr>

## Description 📜

Added ShellSort.md in C_CPP\Algorithmic Approaches\Sorting folder
Fixes #4066

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by aniket-24** [comment]
Please review this...
And merge this PR under Hacktoberfest-accepted Label!

File: C_CPP/Algorithmic Approaches/Sorting/ShellSort.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+1. - SET FLAG = 1, GAP_SIZE = N
+2. - Repeat Steps 3 to 6 while FLAG = 1 OR GAP_SIZE > 1
+3. - SET FLAG = 0
+4. - SET GAP_SIZE = (GAP_SIZE + 1) / 25)
+5. - Repeat Step 6 for I = 0 to I < (N -GAP_SIZE)
+6. - IF Arr[I + GAP_SIZE] > Arr[I]
+     SWAP Arr[I + GAP_SIZE], Arr[I]
+     SET FLAG = 0
+7. - END
+
```
Comment (vasu-1) [inline C_CPP/Algorithmic Approaches/Sorting/ShellSort.md:18]:
```suggestion

- SET FLAG = 1, GAP_SIZE = N
- Repeat Steps 3 to 6 while FLAG = 1 OR GAP_SIZE > 1
- SET FLAG = 0
- SET GAP_SIZE = (GAP_SIZE + 1) / 25)
- Repeat Step 6 for I = 0 to I < (N -GAP_SIZE)
- IF Arr[I + GAP_SIZE] > Arr[I]
     SWAP Arr[I + GAP_SIZE], Arr[I]
     SET FLAG = 0
- END

```

File: C_CPP/Algorithmic Approaches/Sorting/ShellSort.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# Code
+
+```
+// Shell Sort in C++ programming
+
+#include <iostream>
```
Comment (vasu-1) [inline C_CPP/Algorithmic Approaches/Sorting/ShellSort.md:24]:
```suggestion

```c++
// Shell Sort in C++ programming

#include <iostream>
```

File: C_CPP/Algorithmic Approaches/Sorting/ShellSort.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  Average O(nlog n)
+
+- Space Complexity
+
+  O(1)
+
+- Stability
+
+  No
```
Comment (vasu-1) [inline C_CPP/Algorithmic Approaches/Sorting/ShellSort.md:85]: do markdown formatting here !

**Review by vasu-1 — CHANGES_REQUESTED** [review]
- Plagiarisms Detected
- Use your own words don't copy from other resources.
- remove some markdown formatting errors

File: C_CPP/Algorithmic Approaches/Sorting/ShellSort.md (hunk 3 above)
Comment (aniket-24) [reply C_CPP/Algorithmic Approaches/Sorting/ShellSort.md]: okay!

**Comment by aniket-24** [comment]
Now is it okay? @vasu-1

**Comment by vasu-1** [comment]
Good @aniket-24 
Good to go for merge @Suvraneel

**Comment by aniket-24** [comment]
Can you merge this PR under Hacktoberfest-accepted Label!

**Comment by Saviour1001** [comment]
@aniket-24 it gets counted under Hacktoberfest without any labels. Merging this right now. Ping me on Discord if it didn't get counted.

**Pull request merged**